include/SFML/Network/TcpSocket.hpp

```diff
@@ -128,6 +174,339 @@ class SFML_NETWORK_API TcpSocket : public Socket
     ////////////////////////////////////////////////////////////
     void disconnect();
 
+    ////////////////////////////////////////////////////////////
+    /// \brief Set up transport layer security as a client
+    ///
+    /// Once the TCP connection is connected, transport layer
+    /// security can be set up.
+    ///
+    /// All the necessary cryptographic initialization will
+    /// be performed when this function is called.
+    ///
+    /// If this function is called before the TCP connection is
+    /// connected, it will return `TlsStatus::NotConnected` and
+    /// must be called again once the TCP connection is connected.
+    ///
+    /// If this function started TLS setup but could not finish
+    /// it within this call e.g. because this socket was set to
+    /// non-blocking, it will return `TlsStatus::HandshakeStarted`
+    /// and this function will have to be called repeatedly until
+    /// `TlsStatus::HandshakeComplete` is returned. If this socket
+    /// is blocking, `TlsStatus::HandshakeComplete` should be
+    /// returned within the same function call if TLS setup was
+    /// successful.
+    ///
+    /// If `TlsStatus::Error` is returned, something went wrong
+    /// with TLS setup and the connection must be reconnected and
+    /// TLS setup reattempted after it is connected again.
+    ///
+    /// If verification is enabled, this function verifies the peer
+    /// using the system provided certificate store. If the peer
+    /// does not have a certificate that was signed by a certificate
+    /// authority i.e. a self-signed certificate, the entire certificate
+    /// chain can be provided using the alternative overload.
+    ///
+    /// Servers that host multiple services under different names
+    /// need to know which of those services we want to connect
+    /// to in order to reply with the correct certificate chain.
+    /// Server name indication (SNI) is used for this purpose. The
+    /// hostname provided to this function is sent to the server
+    /// if it supports SNI in order for it to return the corresponding
+    /// certificate chain. The hostname is then used to verify the
+    /// certificate chain that was returned by the server. If the
+    /// server does not support SNI or only serves a single
+    /// certificate chain, the hostname will only be used for
+    /// verification.
+    ///
+    /// \param hostname   Hostname of the remote peer, used for verification
+    /// \param verifyPeer `true` to enable peer verification, `false` to disable it
+    ///
+    /// \return TLS status code
+    ///
+    /// \see `setupTlsServer`
+    ///
+    ////////////////////////////////////////////////////////////
+    TlsStatus setupTlsClient(const sf::String& hostname, bool verifyPeer = true);
```

Comment:
if you have an `enum class` instead of a bool, there would not be overload collisions.
something like:
```c++
enum class VerifyPeer
{
    No,
    Yes,
};
```